### PR TITLE
Update devfile schema version

### PR DESCRIPTION
**Pull request by mezarin**
This PR updates the schema version of the various sample devfiles to v2.1.0, it updates the OL driver version used by the devfiles to v21.0.0.9, and updates the maven/gradle Open Liberty plugin versions used.

### CODE REVIEW

File: devfiles/m2-repo-volume/devfile.yaml [hunk 1]
```diff
@@ -42,21 +42,21 @@ parent:
...
-        commandLine: mvn -Dmaven.repo.local=/home/user/.m2/repository -Dliberty.runtime.version=21.0.0.6 -Ddebug=false liberty:dev
+        commandLine: mvn -Dmaven.repo.local=/home/user/.m2/repository -Dliberty.runtime.version=21.0.0.9 -Ddebug=false liberty:dev
     - id: debug
       exec:
-        commandLine: mvn -Dmaven.repo.local=/home/user/.m2/repository -Dliberty.runtime.version=21.0.0.6 -DdebugPort=${DEBUG_PORT} liberty:dev -Dliberty.env.WLP_DEBUG_REMOTE=y
+        commandLine: mvn -Dmaven.repo.local=/home/user/.m2/repository -Dliberty.runtime.version=21.0.0.9 -DdebugPort=${DEBUG_PORT} liberty:dev -Dliberty.env.WLP_DEBUG_REMOTE=y
     - id: test
       exec:
-        commandLine: mvn -Dmaven.repo.local=/home/user/.m2/repository failsafe:integration-test failsafe:verify
+        commandLine: mvn compiler:compile -Dmaven.repo.local=/home/user/.m2/repository failsafe:integration-test failsafe:verify
```
Comment (ajm01) [inline devfiles/m2-repo-volume/devfile.yaml:59]: is the compiler:compile needed to cause the testcases to re-build whenever the test cmd is invoked?

File: devfiles/maven-image/devfile.yaml [hunk 2]
```diff
@@ -89,7 +89,7 @@ commands:
   - id: test
     exec:
       component: dev
-      commandLine: mvn -Dmaven.repo.local=/home/user/.m2/repository failsafe:integration-test failsafe:verify
+      commandLine: mvn compiler:compile -Dmaven.repo.local=/home/user/.m2/repository failsafe:integration-test failsafe:verify
```
Comment (ajm01) [inline devfiles/maven-image/devfile.yaml:92]: same as previous question - just curious

**Review by ajm01 — APPROVED** [review]

File: devfiles/m2-repo-volume/devfile.yaml (hunk 1 above)
Comment (scottkurz) [reply devfiles/m2-repo-volume/devfile.yaml:59]: It's needed to resolve inter-module dependencies within the top-level reactor build, if those deps aren't in the local m2.  (And with liberty:dev, the app-built deps wouldn't be in m2 necessarily)

Comment (scottkurz) [reply devfiles/m2-repo-volume/devfile.yaml:59]: Kind of a Maven quirk that this resolves the problem.. can't even find it documented..but there it is.

**Review by scottkurz — APPROVED** [review]
LGTM

**Pull request merged**